### PR TITLE
Allocate enough space on disk for log_merge

### PR DESCRIPTION
**Pull request by h-shaffer**
Make sure that we have enough free blocks for the log trees to
    be merged once we finish mounting the system

### CODE REVIEW

File: utils/src/parallel_restore.c [hunk 1]
```diff
@@ -1136,6 +1136,7 @@ static spr_err_t prepare_alloc_builders(struct scoutfs_parallel_restore_writer *
 	dprintf("starting prepare with start %llu len %llu\n", wri->meta_start, wri->meta_len);
 
 	skip = ALLOC_BLOCKS + (SCOUTFS_ALLOC_LIST_MAX_BLOCKS * 2);
+	skip += le64_to_cpu(wri->super.total_meta_blocks) / 10;
```
Comment (zabbo) [inline utils/src/parallel_restore.c:1139]: "skip" here is the number of blocks that will be used to store allocation structures.  It reflects the number of blocks written by this routine.  By adding the reservation to skip here you're saying that it won't be available as a free record somewhere (either in list block entries or extent items).  Look at the very next thing done with the skip value, it's subtracted from a free extent that's recorded.   So we want to add it to a free extent somewhere.  The next problem is that you can't just add to any old length.  You need to think about the specific extent value that you're removing with the initial / 10 reservation.  It's the end of the block space.  To return that reservation you have to specifically extend the final extent or create a new one.

File: utils/src/parallel_restore.c [hunk 2]
```diff
@@ -1621,6 +1622,15 @@ spr_err_t scoutfs_parallel_restore_init_slices(struct scoutfs_parallel_restore_w
 					       int nr)
 {
 	u64 total = le64_to_cpu(wri->super.total_meta_blocks);
+	/*
+	 * We have to reserve some space (10%) so there are enough free blocks after
+	 * mounting for log_merge to complete
+	 * XXX could possibly take in an arg determining percentage of space to reserve
```
Comment (zabbo) [inline utils/src/parallel_restore.c:1628]: Can just drop the XXX comment here, there's no reason to call out the universal truth that we could always extend behaviour.  I'd also move the rest of the comment up to a define that contains the reservation percentage which is then used in place of the two raw instances of 10.  And 10 is probably too much, maybe 5?

**Review by zabbo — CHANGES_REQUESTED** [review]

File: utils/src/parallel_restore.c (hunk 1 above)
Comment (h-shaffer) [reply utils/src/parallel_restore.c:1139]: What would the benefit be for adding a new extent onto the end vs extending the final extent?

Comment (zabbo) [reply utils/src/parallel_restore.c:1139]: If you extend an existing extent you don't have to create a new one, but you have to identify the extent to extend.

Comment (h-shaffer) [reply utils/src/parallel_restore.c:1139]: Could you elaborate a bit more on identifying an extent, I see that there is some math identifying its bounds, start and len, would I just be able to calculate the end of the reserved space and add onto it?

Comment (h-shaffer) [reply utils/src/parallel_restore.c:1139]: Could you elaborate a bit more on identifying an extent, I see there's some math there identifying its bounds, couldn't I just calculate the end of the free extent and add a percentage of blocks into the extent?

File: utils/src/parallel_restore.c [hunk 3]
```diff
@@ -1146,6 +1154,13 @@ static spr_err_t prepare_alloc_builders(struct scoutfs_parallel_restore_writer *
 	if (err)
 		goto out;
 	wri->meta_len -= len;
+	/* store reserved extra space a free extent*/
+	end = start + len;
+	extra = le64_to_cpu(wri->super.total_meta_blocks) / EXTRA_BLOCK_SLICE;
+	err = insert_free_items(&wri->meta_btb[0], end, extra);
```
Comment (aversecat) [inline utils/src/parallel_restore.c:1160]: could just write `start + len` inside the function call here and avoid needing to declare `end` entirely and pre-calculate.

**Comment by aversecat** [comment]
I've merged the two relevant changes into my tree.

**Pull request closed without merging**